JavaScriptTasks/Session 5 - OOPInheritance(5-3-2020)/ArrayHelpers.js

```diff
@@ -0,0 +1,62 @@
+// PROBLEM STATEMENT :Array Helpers
+
+// http://www.codewars.com/kata/array-helpers
+// SOLUTION:
+
+  // TODO
+//var numbers = [1, 2, 3, 4, 5];
+Array.prototype.square = function(){
+return this.map(function(n){ return n*n;});
+ }
+ 
+ Array.prototype.cube = function(){
+return this.map(function(n){ return n*n*n;});
+ }
+ 
+ Array.prototype.average = function(array){
+ let sum = 0 ,counter = 0 ;
+ this.forEach(function(n){ 
+sum +=n;
+counter++ ;
+});
+
+return sum/counter;
+ }
+ 
+ 
+ Array.prototype.sum = function(){
+ let sum = 0  ;
+ this.forEach(function(n){ 
```

Comment:
instead of forEach please use reduce method

Reply:
revised code as : 

if (!Array.hasOwnProperty('sum')) {
  Array.prototype.sum = function () {
    let sum = 0;
    sum = this.reduce(add);
    return sum;
  }
}
function add(total, num) {
  return total + num;
}